src/core/application/mod.rs

```diff
@@ -0,0 +1,179 @@
+//! The Application is the core component of crius.
```

Comment:
what does `mod` stand for?

Reply:
To explain what this `mod.rs` file is I will have to briefly explain how modules work in Rust.

So each file in rust is a module. For exaple if I had a file named `foo.rs`, any type/function declared in the file will be under the `foo` module.

**foo.rs**:
```rust
// zero sized struct
pub struct Foo;
```
**main.rs**:
```rust
// declare the module so that it is in the compile path
mod foo;

// use a type from the module
use foo::Foo;

fn main() {
   let a = Foo;
}
```

Another way to define a module is within the same file as an embedded module:

**foo.rs**:
```rust
pub struct Foo;

pub mod Bla {
    pub struct Bla;
}
```
**main.rs**:
```rust
// declare the module so that it is in the compile path
mod foo;

// use a type from the module
use foo::{Foo, bla::Bla};

fn main() {
   let a = Foo;

   let b = Bla;
}
```

Now lets say I do not want to use embeded modules, but I actually want to organize my code using separate files but I still want module `bla` to be under module `foo`. I need to create a directory and inside it put a file named `bla.rs` to act as its own module. Something like this:

```
foo
|--- bla.rs
|--- <source_code_for_foo>???
```

However this does not work because Rust has no way of knowing the the directory **foo** should be treated as a module also where to put the source code for the `Foo` struct? 

To remedy that Rust looks inside a directory for a file named `mod.rs`. If this file exists, it treats the entire directory as a module and any other files that exist within it as submodules like so:

```
foo
|--- bla.rs // submodule, must be declared in mod.rs to be compiled
|--- mod.rs //main module
```

So now if I were to implement the same code as above the definition of the `Foo` struct will have to exists inside the `foo` module which now is the `mod.rs` file:

**mod.rs**:
```rust
// declare the submodule so that it is in the compilation path
pub mod bla;

pub struct Foo;
```

**bla.rs**:
```rust
pub struct Bla;
```

**main.rs**:
```rust
// declare the module so that it is in the compile path
mod foo;

// use a type from the module
use foo::{Foo, bla::Bla};

fn main() {
   let a = Foo;

   let b = Bla;
}
```

In this case I wanted the `settings` module to be under the `application` module so that it can be used in code like so:

```rust
use crius::application::settings::Settings;
```

I hope it's a bit more clear now. :)

For a more detailed explanations on modules check the module chapter in the Rust [book](https://doc.rust-lang.org/book/ch07-02-defining-modules-to-control-scope-and-privacy.html).

Reply:
Thanks, this has been very useful. I read [Rust Essentials](https://www.goodreads.com/book/show/25644753-rust-essentials) some time ago, and it wasn't great. I'll read the official book :+1: